gammapy/spectrum/make.py

```diff
@@ -0,0 +1,220 @@
+import logging
+from astropy import units as u
+from astropy.coordinates import Angle
+from astropy.utils import lazyproperty
+from regions import CircleSkyRegion
+from gammapy.irf import EnergyDependentMultiGaussPSF, apply_containment_fraction
+from gammapy.maps import WcsGeom
+from gammapy.maps.geom import frame_to_coordsys
+from .core import CountsSpectrum
+from .dataset import SpectrumDataset
+
+log = logging.getLogger(__name__)
+
+
+class SpectrumDatasetMaker:
+    """Make spectrum for a single IACT observation.
+
+    The irfs and background are computed at a single fixed offset,
+    which is recommend only for point-sources.
+
+    Parameters
+    ----------
+    region : `~regions.SkyRegion`
+        Region to compute spectrum dataset for.
+    e_reco : `~astropy.units.Quantity`
+        Reconstructed energy binning
+    e_true : `~astropy.units.Quantity`
+        True energy binning
+    containment_correction : bool
+        Apply containment correction for point sources and circular on regions.
+    binsz : `~astropy.coordinates.Angle`
+        Reference map bin size.
+    width : `~astropy.coordinates.Angle`
+        Reference map width, should encompass the whole region.
+
+    """
+
+    def __init__(
+        self,
+        region,
+        e_reco,
+        e_true=None,
+        containment_correction=True,
+        binsz="0.01 deg",
+        width="0.5 deg",
```

Comment:
Technically it would be better if `binsz` is also defined w.r.t. the region size since it will impact strongly the estimate of the solid angle.

Reply:
I decided to only support circular regions in `make_background` and hard coded to the analytical solution for the solid angle. This way I could also remove the `binsz` and `width` parameters, because for the event selection it does not matter. 

This seemed to me the best compromise for now. In case people want to use reflected regions with different region shapes, they are not interested in the background spectrum anyway and can just disable it in `SpectrumDatasetMaker.run()`.